src/main/java/codeu/controller/ChatServlet.java

```diff
@@ -76,6 +77,33 @@ void setUserStore(UserStore userStore) {
     this.userStore = userStore;
   }
 
+  public static String replaceUrls(String text){
+    String newText = text;
+    ArrayList<String> containedUrls = new ArrayList<String>();
+    Pattern pattern = Pattern.compile(
+       "(?:^|[\\W])((ht|f)tp(s?):\\/\\/|www\\.)"
+                + "(([\\w\\-]+\\.){1,}?([\\w\\-.~]+\\/?)*"
+                + "[\\p{Alnum}.,%_=?&#\\-+()\\[\\]\\*$~@!:/{};']*)",
+                Pattern.CASE_INSENSITIVE | Pattern.MULTILINE | Pattern.DOTALL);
+    Matcher urlMatcher = pattern.matcher(text);
+
+    while (urlMatcher.find())
+    {
+        containedUrls.add(text.substring(urlMatcher.start(0),
+                urlMatcher.end(0)));
+    }
+
+    // iterate backwards over containedUrls to update the text with <a></a> tags
+    for (int i = containedUrls.size()-1; i >=0; i--){
+      int index = text.lastIndexOf(containedUrls.get(i));
+      //remove this instance of the url from the original text:
+      text = text.substring(0,index) + text.substring(index + containedUrls.get(i).length());
+      //update the instance of the url in the new text:
+      newText = newText.substring(0,index) + "<a href='" + containedUrls.get(i) + "'>" + containedUrls.get(i) + "</a>" + newText.substring(index + containedUrls.get(i).length());
```

Comment:
I would recommend breaking this very long line to make it more readable

Reply:
I tried looking for this to edit it, but I could not find it, did someone already fix this readability issue?

Reply:
This hasn't been merged into master yet, so you would only be able to see it if you were on the urlstyledtext branch